### PR TITLE
Add Joke Controller and Tests

### LINKED ISSUE #7: Add Joke Controller and Tests
# Acceptance Criteria:

- [ ] There is <tt>JokeController.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] <tt>JokeController.java</tt> implements a class that implements an API endpoint that takes in the necessary
      parameters, calls the <tt>JokeQueryService</tt>, and returns the results.
- [ ] The file  <tt>JokeController.java</tt> file make appropriate use of the 
      annotations `@Tag`, `@Operation` and `@Parameter` to document the API.
- [ ] The controller endpoint shows up on the Swagger UI page.
- [ ] There is a <tt>JokeControllerTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] The tests in <tt>JokeControllerTests</tt> provide 100% coverage of <tt>JokeController</tt> for both jacoco and pitest.
      
# Details

See the assignment description at https://ucsb-cs156.github.io/f23/lab/team01 for details.

### PR DESCRIPTION
**Pull request by tigeryu8900**
In this PR, we add an endpoint `/api/jokes/get` that can be used to get a certain amount of jokes of a particular category.

Closes #7

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java [hunk 1]
```diff
@@ -1,8 +1,34 @@
...
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name="College Subreddits from https://github.com/karlding/college-subreddits/")
```
Comment (MuMuKow) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java:16]: Modify description

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java [hunk 2]
```diff
@@ -1,8 +1,34 @@
...
 @RestController
+@RequestMapping("/api/jokes")
 public class JokeController {
-    
+    ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    JokeQueryService jokeQueryService;
+
+    @Operation(summary = "Get a list of college subreddits")
```
Comment (MuMuKow) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java:25]: Modify description

**Review by MuMuKow — CHANGES_REQUESTED** [review]
Please address following changes

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java (hunk 1 above)
Comment (tigeryu8900) [reply src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java]: Fixed

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java (hunk 2 above)
Comment (tigeryu8900) [reply src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/JokeController.java]: Fixed

**Review by MuMuKow — APPROVED** [review]
LGTM

**Pull request merged**